### ISSUE TITLE
deskew confidence?

### ISSUE THREAD

**Issue by DiTo97** [sbrunner/deskew#413]
Hi @sbrunner,

Is there a way to retrieve the deskew confidence?

It would make it much easier to avoid rotating the image when the confidence in the skew angle is low.

**Comment by sbrunner** [comment]
Good question,

Currently, there is nothing like it, but it will be a great improvement :-)

**Comment by DiTo97** [comment]
Do you have any pointers on how such a confidence should be implemented?

I may imagine something like the rate of adherence of the maximum peak's frequency against that of the other peaks...

**Comment by sbrunner** [comment]
Or an addition of the dist on the same angle?
https://github.com/sbrunner/deskew/blob/master/deskew/__init__.py#L48

**Comment by DiTo97** [comment]
> Or an addition of the dist on the same angle? https://github.com/sbrunner/deskew/blob/master/deskew/__init__.py#L48

Do you mean summing the distances for each angle instead of [summing ones](https://github.com/sbrunner/deskew/blob/master/deskew/__init__.py#L59) as you do in the `freqs` array? Then, softmax?

What is the rationale behind summing distances and confidence, tho?

**Comment by sbrunner** [comment]
Yes, if the line concerned is long, it's more confident than if it's short, not?

**Comment by DiTo97** [comment]
I just don't see how the $\rho$ parameters (the `dists` array) of the lines parameterized as $(\rho, \theta)$ in Hough space, or their magnitude (short/long) would relate to confidence

I am probably just lacking some Hough transform knowledge...
